app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt

```diff
@@ -0,0 +1,197 @@
+                    HomeRoute(
+                        onNavigateToMovieDetails = { movieId ->
+                            navController.navigate(
+                                NavigationItem.MovieDetailsDestination.createNavigationRoute(movieId)
+                            )
+                            showBottomBar = false
+                        }
+                    )
+                }
+                composable(NavigationItem.FavoritesDestination.route) {
+                    showBottomBar = true
+                    showBackIcon = showBackIcon
+                    FavoritesRoute(
+                        onNavigateToMovieDetails = { movieId ->
+                            navController.navigate(
+                                NavigationItem.MovieDetailsDestination.createNavigationRoute(movieId)
+                            )
+                            showBottomBar = false
+                        }
+                    )
+                }
+                composable(
+                    route = NavigationItem.MovieDetailsDestination.route,
+                    arguments = listOf(navArgument(MOVIE_ID_KEY) { type = NavType.IntType }),
+                ) {
+                    MovieDetailsRoute()
+                    showBottomBar = false
+                    showBackIcon = showBackIcon
+                }
+            }
+        }
+    }
+}
+
+@ExperimentalMaterial3Api
+@Composable
+private fun TopBar(
+    navigationIcon: @Composable (() -> Unit)? = null
+) {
+    CenterAlignedTopAppBar(
+         title = {
+             Image(
+                 painter = painterResource(id = R.drawable.ic_logo),
+                 contentDescription = null
+             )
+         },
+         navigationIcon = {
+             navigationIcon?.invoke()
+         },
+        colors = TopAppBarDefaults.centerAlignedTopAppBarColors(
+            containerColor = colorResource(id = R.color.top_bar)
+        )
+     )
+}
+
+@Composable
+private fun BackIcon(
+    onBackClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Image(
+        painter = painterResource(id = R.drawable.ic_arrow_back),
+        contentDescription = stringResource(id = R.string.back),
+        modifier = modifier
+            .padding(start = dimensionResource(id = R.dimen.padding_medium))
+            .clickable { onBackClick() }
+    )
+}
+
+@Composable
+private fun BottomNavigationBar(
+    destinations: List<NavigationItem>,
+    onNavigateToDestination: (NavigationItem) -> Unit,
+    currentDestination: NavDestination?,
+) {
+    BottomNavigation(
+        backgroundColor = MaterialTheme.colors.background
+    ) {
+        destinations.forEach { destination ->
+            AddItem(
+                destination = destination,
+                currentDestination = currentDestination,
+                onNavigateToDestination = onNavigateToDestination
+            )
+        }
+    }
+}
+
+@Composable
+fun RowScope.AddItem(
```

Comment:
Should just be called `Item` or `BottomNavItem` or something similar.
Composable functions represent a UI component and not an action to be performed (AddItem -> Item).